daemon/swap_daemon.go

```diff
@@ -0,0 +1,165 @@
+// Package daemon is responsible for assembling, running and cleanly shutting
+// down the swap daemon (swapd) and its numerous subcomponents.
+package daemon
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"path"
+
+	"github.com/ChainSafe/chaindb"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/hashicorp/go-multierror"
```

Comment:
Apparently, multierror is so popular that Go 1.20 will have a version of it integrated into the language.